src/modules/esl-base-element/decorators/attr.ts

```diff
@@ -1,25 +1,33 @@
 import {toKebabCase} from '../../esl-utils/misc/format';
+import {identity} from '../../esl-utils/misc/functions';
+
 import type {ESLBaseElement} from '../core/esl-base-element';
+import type {Serializer} from './serializers';
 
 /** HTML attribute mapping configuration */
-type AttrDescriptor = {
+type AttrDescriptor<T> = {
   /** HTML attribute name. Uses kebab-cased variable name by default */
   name?: string;
   /** Create getter only */
   readonly?: boolean;
   /** Use data-* attribute */
   dataAttr?: boolean;
   /** Default property value. Used if no attribute is present on the element. Empty string by default. */
-  defaultValue?: string | boolean | null;
+  defaultValue?: T ;
```

Comment:
```suggestion
  defaultValue?: T;
```

Reply:
thanks for your attention, I’ll fix it up